include/ibamr/LevelSetUtilities.h

```diff
@@ -0,0 +1,420 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (c) 2017 - 2020 by the IBAMR developers
+// All rights reserved.
+//
+// This file is part of IBAMR.
+//
+// IBAMR is free software and is distributed under the 3-clause BSD
+// license. The full text of the license can be found in the file
+// COPYRIGHT at the top level directory of IBAMR.
+//
+// ---------------------------------------------------------------------
+
+/////////////////////////////// INCLUDE GUARD ////////////////////////////////
+
+#ifndef included_IBAMR_LevelSetUtilities
+#define included_IBAMR_LevelSetUtilities
+
+/////////////////////////////// INCLUDES /////////////////////////////////////
+
+#include <ibamr/config.h>
+
+#include "ibamr/LSInitStrategy.h"
+
+#include "tbox/DescribedClass.h"
+#include "tbox/Pointer.h"
+#include "tbox/Serializable.h"
+
+#include <limits>
+#include <string>
+
+namespace SAMRAI
+{
+namespace hier
+{
+template <int DIM>
+class PatchHierarchy;
+}
+namespace pdat
+{
+template <int DIM, class TYPE>
+class CellVariable;
+} // namespace pdat
+namespace tbox
+{
+class Database;
+} // namespace tbox
+} // namespace SAMRAI
+
+namespace IBAMR
+{
+class AdvDiffHierarchyIntegrator;
+} // namespace IBAMR
+
+namespace IBTK
+{
+class HierarchyMathOps;
+}
+
+/////////////////////////////// CLASS DEFINITION /////////////////////////////
+
+namespace IBAMR
+{
+/*!
+ * \brief Class LevelSetUtilities provides implementation of some helper functions
+ * to enforce mass/volume conservation of phases.
+ */
+
+class LevelSetUtilities
```

Comment:
@drwells do you have an opinion here about using inner classes vs. classes in a namespace? I feel like we do not do a lot of inner classes, although of course there are some.

Reply:
Also, should the function implementations go in an `-inl.h` file in `include/ibamr/private`?

Reply:
I think we talked about this in-person at some point: lets start transitioning things to being in namespaces.